src/components/pages/Header.jsx

```diff
@@ -0,0 +1,57 @@
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+import DesktopNav from "../navbars/DesktopNav";
+import MobNav from "../navbars/MobNav";
+import ProfileNav from "../navbars/ProfileNav";
+
+const Header = () => {
+    const [isSearch, setIsSearch] = useState(true);
+    const [isLoginBtn, setIsLoginBtn] = useState(true);
+    const [isJoinBtn, setIsJoinBtn] = useState(true);
+    
+    const location = useLocation();
+    useEffect(() => {
+        if (location.pathname === "/search") {
+            setIsSearch(false);
+        } else {
+            setIsSearch(true);
+        }
+        
+        if(location.pathname === "/login") {
+            setIsLoginBtn(false);
+        } else {
+            setIsLoginBtn(true);
+        }
+        
+        if(location.pathname === "/register") {
+            setIsJoinBtn(false);
+        } else {
+            setIsJoinBtn(true);
+        }
+    }, [location.pathname]);
+
+    const navigate = useNavigate();
+    const search = () => {
+        navigate("/search");
```

Comment:
There is a single expressions in these functions. These can benefit from being turned into one liners for clarity.